pkg/packaging/packaging.go

```diff
@@ -92,59 +92,54 @@ func (p *PackageOptions) Build(ctx context.Context, packageWriter io.Writer, tar
 		return errors.Wrap(err, "setup directories")
 	}
 
-	launcherEnv := map[string]string{
-		"KOLIDE_LAUNCHER_HOSTNAME":           p.Hostname,
-		"KOLIDE_LAUNCHER_ROOT_DIRECTORY":     p.rootDir,
-		"KOLIDE_LAUNCHER_OSQUERYD_PATH":      filepath.Join(p.binDir, "osqueryd"),
-		"KOLIDE_LAUNCHER_ENROLL_SECRET_PATH": filepath.Join(p.confDir, "secret"),
+	flagFilePath := filepath.Join(p.confDir, "launcher.flags")
+	flagFile, err := os.Create(filepath.Join(p.packageRoot, flagFilePath))
+	if err != nil {
+		return errors.Wrap(err, "creating flag file")
+	}
+	defer flagFile.Close()
+
+	launcherMapFlags := map[string]string{
+		"hostname":           p.Hostname,
+		"root_directory":     p.canonicalizePath(p.rootDir),
+		"osqueryd_path":      p.canonicalizePath(filepath.Join(p.binDir, "osqueryd")),
+		"enroll_secret_path": p.canonicalizePath(filepath.Join(p.confDir, "secret")),
 	}
 
-	launcherFlags := []string{}
+	launcherBoolFlags := []string{}
 
 	if p.InitialRunner {
-		launcherFlags = append(launcherFlags, "--with_initial_runner")
+		launcherBoolFlags = append(launcherBoolFlags, "with_initial_runner")
 	}
 
 	if p.Control && p.ControlHostname != "" {
-		launcherEnv["KOLIDE_LAUNCHER_CONTROL_HOSTNAME"] = p.ControlHostname
+		launcherMapFlags["control_hostname"] = p.ControlHostname
 	}
 
 	if p.Autoupdate && p.UpdateChannel != "" {
-		launcherFlags = append(launcherFlags, "--autoupdate")
-		launcherEnv["KOLIDE_LAUNCHER_UPDATE_CHANNEL"] = p.UpdateChannel
+		launcherBoolFlags = append(launcherBoolFlags, "autoupdate")
+		launcherMapFlags["update_channel"] = p.UpdateChannel
 	}
 
 	if p.CertPins != "" {
-		launcherEnv["KOLIDE_LAUNCHER_CERT_PINS"] = p.CertPins
+		launcherMapFlags["cert_pins"] = p.CertPins
 	}
 
 	if p.DisableControlTLS {
-		launcherFlags = append(launcherFlags, "--disable_control_tls")
+		launcherBoolFlags = append(launcherBoolFlags, "disable_control_tls")
 	}
 
 	if p.InsecureGrpc {
-		launcherFlags = append(launcherFlags, "--insecure_grpc")
+		launcherBoolFlags = append(launcherBoolFlags, "insecure_grpc")
```

Comment:
Are we going to want to support the `-insecure_jsonrpc` flag?

Reply:
Erm. Yes, but I feel like we might want to merge `-insecure_grpc` and `-insecure_jsonrpc` into `insecure_transport`. Thoughts?

Reply:
Yea that makes sense I think

Reply:
Though, I'd probably rather re-plumb that in a separate PR

Reply:
yea it can go in another PR